node/src/components/consensus/era_supervisor.rs

```diff
@@ -1167,16 +1172,45 @@ where
                 }
                 self.era_mut(era_id)
                     .add_candidate(candidate_block, missing_evidence);
+                let proto_block_deploys_set: BTreeSet<DeployHash> =
+                    proto_block.deploys_iter().cloned().collect();
+                for ancestor_block in ancestor_blocks {
+                    let ancestor_proto_block = ancestor_block.proto_block();
+                    for deploy in ancestor_proto_block.deploys_iter() {
+                        if proto_block_deploys_set.contains(deploy) {
+                            effects.extend(self.resolve_validity(
+                                era_id,
+                                sender,
+                                proto_block,
+                                timestamp,
+                                false,
+                            ));
+                            return effects;
+                        }
+                    }
+                }
```

Comment:
Should we even ask for resolving the validity for deploys we know are replay attacks?

Reply:
It's not asking, it's _communicating_ the invalidity (the `false` argument is the answer to the question "is it valid?").

Reply:
Ah, it always trips me. The name is wrong. It's not asking _to resolve validity_, it signals that the _validity is resolved_.